controllers/main.go

```diff
@@ -1399,13 +1405,76 @@ func (controller *MainController) Stats(c web.C, r *http.Request) (string, int)
 func (controller *MainController) Status(c web.C, r *http.Request) (string, int) {
 	var rpcstatus = "Running"
 
-	if controller.RPCIsStopped() {
-		rpcstatus = "Stopped"
+	// Confirm that the incoming IP address is an approved
+	// admin IP as set in config.
+	remoteIP := r.RemoteAddr
+	if strings.Contains(remoteIP, ":") {
+		parts := strings.Split(remoteIP, ":")
+		remoteIP = parts[0]
+	}
+
+	if !stringSliceContains(controller.adminIPs, remoteIP) {
+		return "/error", http.StatusSeeOther
+	}
+
+	// Attempt to query wallet statuses
+	walletInfo, err := controller.WalletStatus()
+	if err != nil {
+		// decide when to throw err here
+	}
+
+	type WalletInfoPage struct {
+		Connected         bool
+		DaemonConnected   bool
+		Unlocked          bool
+		TicketMaxPrice    float64
+		BalanceToMaintain float64
+		StakeMining       bool
+	}
+	walletPageInfo := make([]WalletInfoPage, len(walletInfo), len(walletInfo))
+	connectedWallets := 0
+	for i, v := range walletInfo {
+		// If something is nil in the slice means it is disconnected.
+		if v == nil {
+			walletPageInfo[i] = WalletInfoPage{
+				Connected: false,
+			}
+			controller.rpcServers.DisconnectWalletRPC(i)
+			err = controller.rpcServers.ReconnectWalletRPC(i)
+			if err != nil {
+				log.Infof("wallet rpc reconnect failed: server %v %v", i, err)
+			}
+			continue
+		}
+		// Wallet has been successfully queried.
+		connectedWallets++
+		walletPageInfo[i] = WalletInfoPage{
+			Connected:         true,
+			DaemonConnected:   v.DaemonConnected,
+			Unlocked:          v.Unlocked,
+			TicketMaxPrice:    v.TicketMaxPrice,
+			BalanceToMaintain: v.BalanceToMaintain,
+			StakeMining:       v.StakeMining,
+		}
+	}
+
+	// Depending on how many wallets have been detected update RPCStatus.
+	// Admins can then use to monitor this page periodically and check status.
+	switch connectedWallets {
+	case 0:
+		rpcstatus = "Emergency!"
+	case 1:
+		rpcstatus = "Critical"
+	case 2:
+		rpcstatus = "Degraded"
```

Comment:
Not really sure what to do about this now that minservers was added. Maybe something like:

```go
rpcstatus := "Unknown"
allWallets := len(walletInfo)

if connectedWallets == allWallets {
 rpcstatus = "OK"
} else {
  if connectedWallets == 0 {
   rpcstatus = "Emergency"
  } else if connectedWallets == 1 {
   rpcstatus = "Critical"
  } else {
   rpcstatus = "Degraded"
  }
}
```